### PR TITLE
Anti-mob lag nuclear option

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request
Locks behind most of superior mob life behind an ai_inactive check.

This is a nuclear option. Dead mobs will no longer process reagents among other things. Mobs will become more buggy. I did not test this much because I am getting really tired of this codebase. 

This WILL reduce mob lag. I dont know if its worth it.

## Changelog
:cl:
nuclear option: Mobs no longer process if their AI is inactive.
/:cl:

### CODE REVIEW

**Comment by cdb-is-not-good** [comment]
I'll test this myself likely after the weekend to see how functional it is.

File: code/modules/mob/living/carbon/superior_animal/superior_animal.dm [hunk 1]
```diff
@@ -550,49 +550,50 @@
...
+			ticks_processed = 0
+		if(handle_cheap_regular_status_updates()) // They have died after all of this, do not scan or do not handle AI anymore.
+			return PROCESS_KILL
+
+		if (can_burrow && bad_environment)
+			evacuate()
+
+		if (!weakened)
+
+			if(!AI_inactive) //we dont need to handle ai if we're disabled
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/superior_animal/superior_animal.dm:584]: Why have this check twice?

**Comment by Trilbyspaceclone** [comment]
This isnt a nuclear option imo, i think its a good pr

File: code/modules/mob/living/carbon/superior_animal/superior_animal.dm (hunk 1 above)
Comment (nikothedude) [reply code/modules/mob/living/carbon/superior_animal/superior_animal.dm:584]: because i made this pr in 5 minutes

**Comment by nikothedude** [comment]
This isnt needed. Ill do a better version of this later.

**Pull request closed without merging**